docs/solidity-by-example.rst

```diff
@@ -619,4 +619,63 @@ Safe Remote Purchase
 Micropayment Channel
 ********************
 
-To be written.
+::
+
+    pragma solidity ^0.4.11;
+
+    contract PaymentChannel {
+        address initiator;
+        address acceptor;
+        uint deposit;
+        uint timeout;
+        uint closeTimestamp;
+        uint closeSequenceNumber;
+        uint closeBalanceInitiator;
+
+        enum State { Opened, Accepted, Closing, Closed }
+        State state;
+
+        function PaymentChannel(address _other, uint _timeout) payable {
+            initator = msg.sender;
+            acceptor = _other;
+            deposit = msg.value;
+            require(deposit < 2**250);
```

Comment:
I'm curious about this overflow check on msg.value. I wouldn't think to do that... is it really necessary?

Reply:
Not sure why I did this. Perhaps better safe than sorry?